### DISCUSSION TITLE
Upgrading Appwrite while running Docker as a non-root user (Rootless mode)

### DISCUSSION THREAD

**Discussion by a-l-e-c** [appwrite/appwrite#5557] [Q&A]
The goal here is to document the steps for others, or hopefully get some pointers on what could be changed/improved to prevent the additional steps I have to do to upgrade Appwrite while running docker in "Rootless mode".

First a little background on my Docker setup:
1. I'm running Docker in [Rootless mode](https://docs.docker.com/engine/security/rootless/)
2. Docker version is `20.10.22`
3. Docker Compose version `v2.14.1`
4. I initially installed Appwrite using `https://appwrite.io/install/compose`, and only had to modify the `docker.sock`
```
sed -i "s+/var/run/docker.sock:+/run/user/${UID}/docker.sock:+g" docker-compose.yml
```

I've been running Appwrite (`v:1.0.3.501`) for a few months now, but finally decided to create my backups and attempt to upgrade to `v1.1.0`.
The following will install Appwrite succesfully, but will not run without an error (due to the wrong value set for `docker.sock` in the new `docker-compose.yml`)
```
docker run -it --rm \
    --volume $XDG_RUNTIME_DIR/docker.sock:/var/run/docker.sock \
    --volume "$(pwd)"/appwrite:/usr/src/code/appwrite:rw \
    --entrypoint="install" \
    appwrite/appwrite:1.1.0
```

After that, I need to stop Appwrite and apply the fix:
```
cd ~/appwrite
docker compose stop
sed -i "s+/var/run/docker.sock:+/run/user/${UID}/docker.sock:+g" docker-compose.yml
docker compose -f docker-compose.yml up -d --remove-orphans --renew-anon-volumes
```

_FYI: I've upgraded up to v1.1.2 and will proceed further during the following days._

**Comment by moeep** [comment]
I have tried some things, just like you, and it seems that someone in the Discord thread had similar problems as described here. Here is a brief description of what currently causes difficulties for me during the major update from version 1.3.8 to 1.4.2 [](https://discord.com/channels/564160730845151244/1147674771195183134).

In this context, we may have to accept that, at least in the rootless variant, manual changes to the files docker-compose.yml and .env are necessary. At least, this has led to the successful execution of the migration command.

I wanted to mention this here again so that others don't have to search endlessly for solutions.

Edit:

A small update that might be of interest to others: It seems that either a lot has changed in the Appwrite update process or it simply required the right approach. I now use the following command and define $DOCKER_HOST_ROOTLESS_PATH in my .env file as a variable to ensure the smooth startup of Appwrite under Docker in Rootless mode:

docker run -it --rm \
    --volume $DOCKER_HOST_ROOTLESS_PATH:/var/run/docker.sock \
    --volume "$(pwd)"/appwrite:/usr/src/code/appwrite:rw \
    --entrypoint="upgrade" \
    appwrite/appwrite:1.4.13

It's important to make sure you are in the directory that acts as the parent directory of the Appwrite version. This way, Appwrite can find the required version and create backups of the docker-compose.yml and .env files. Afterward, you'll need to transfer your own customizations back into these files, such as, in my case, the use of $DOCKER_HOST_ROOTLESS_PATH in Docker Compose and .env.

I still recommend creating manual backups, as this is a proven method to ensure you can revert to the previous configuration in case of any issues. The upgrade from Version 11 to Version 13 appears to have gone smoothly, which is great. It might have been a combination of adjustments and a bit of luck.
